File: vigenere/vigenere.py
import cs50
import sys

def vigenere(plaintext, key):

    cyphertext = []

    key_length = len(key) #this is the wrapper
    plaintext_list = range(len(plaintext))

    for i,j in zip(plaintext, plaintext_list):

        # accepts uppercase
        if ord(i)>=65 and ord(i)<(65+25):
            cypher_char = chr( ord(i)+(j%key_length) )
            cyphertext.append(cypher_char)

        # accepts lowercase
        elif ord(i)>=97 and ord(i)<(97+25):
            cypher_char = chr( ord(i)+(j%key_length) )
            cyphertext.append(cypher_char)

        else:
            cyphertext.append(i)

    return cyphertext

def main():

    if len(sys.argv) != 2:
        print("argument needed: key (vigenere encryption)")
        exit(1)

    key = sys.argv[1]

    print("Plaintext: ", end=" ")
    plaintext = cs50.get_string()

    vigenere_encryption = vigenere(plaintext, key)
    print(vigenere_encryption)

    exit(0)


if __name__ == "__main__":
    main()

File: vigenere/vigenere.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

void vigenere(string plaintext, string key);

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./vigenere key\n");
        return 1;
    }

    string key = argv[1];
    printf("plaintext: ");
    string plaintext = get_string();
    
    vigenere(plaintext, key);
    
    printf("\n");
    return 0;
    
}

void vigenere(string plaintext, string key)
{
    //int cap_ascii = 65;
    //int low_ascii = 97;
    
    printf("cyphertext: ");
    if (plaintext != NULL)
    {
        for (int i=0, n=strlen(plaintext); i<n; i++)
        {
            if (isupper(plaintext[i]))
            {
                // make the key cyclic: (char) ((i) % strlen(key))+cap_ascii
                // outputs the cyphertext = plaintext + cyclic key
                printf("%c", (char) ( plaintext[i]+(i%strlen(key)) ));
            }
            else if (islower(plaintext[i]))
            {
                // outputs the cyphertext = plaintext + cyclic key
                printf("%c", (char) ( plaintext[i]+(i%strlen(key)) ));
            }
            else
            {
                printf("%c", plaintext[i]);
            }
        }
    }
}

